src/lib/io/include/sol-uart.h

```diff
@@ -81,22 +90,64 @@ enum sol_uart_parity {
  * @brief Amount of stop bits.
  */
 enum sol_uart_stop_bits {
-    SOL_UART_STOP_BITS_ONE = 0,
-    SOL_UART_STOP_BITS_TWO
+    SOL_UART_STOP_BITS_ONE = 0, /**< Use one stop bit*/
+    SOL_UART_STOP_BITS_TWO /**< Use two stop bits */
 };
 
+/**
+ * @struct sol_uart_config
+ * @brief A configuration struct used to set the UART paramenters.
+ *
+ * @see sol_uart_open()
+ * @note UART follows the Soletta stream guidelines, which can be found here: @ref streams
```

Comment:
`s/guidelines/design patterns/g`, that is, all the code. Keep it consistent, since the page and samples are named like that.